controllers/discovery/initDiscoveryUserData.js

```diff
@@ -50,6 +50,8 @@ const InitDiscoveryUserData = async (req, res) => {
     );
     let topicIds = user_topics.map((topic) => topic.topic_id);
 
+    const topicIdQuery = topicIds?.length > 0 ? `AND tp.topic_id in (:topicIds)` : '';
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of optional chaining (`?.`) is unnecessary here because `map` always returns an array, so `topicIds` will never be `null` or `undefined`. It can be safely removed to simplify the code.
```diff
- const topicIdQuery = topicIds?.length > 0 ? `AND tp.topic_id in (:topicIds)` : '';
+ const topicIdQuery = topicIds.length > 0 ? `AND tp.topic_id in (:topicIds)` : '';
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->